### DISCUSSION TITLE
selected entry callback

### DISCUSSION THREAD

**Discussion by xzbdmw** [ibhagwan/fzf-lua#1135] [Q&A] [answered]
for lsp_document_symbols, I want to auto jump to location and highlight code ranges as I <C-n> through fzf result list rather than look at preview windows, could you give me some hint where the callback get executed in source code?

**Comment by ibhagwan** [comment, marked as answer]
LSP symbols uses the same previewer as any file entry previewer, it displays file, line and column.

If you want to do something different look the advanced Wiki, create your own previewer and do whatever you want with the callback:
https://github.com/ibhagwan/fzf-lua/wiki/Advanced#preview-nvim-builtin

**Reply by xzbdmw** [reply]
Get it, thanks a lot.

**Reply by xzbdmw** [reply]
Hmm, how could I get range information in previewer functions

**Reply by ibhagwan** [reply]
Unfortunately the builtin previewer isn’t well documented but fortunately lua is very simple, clone the code and explore, it’s fairly simple.

**Reply by xzbdmw** [reply]
Yes, I'm trying to find the place where preview(each time I change selection) is called, so I can inject some code there, but the callback is hard to track down

**Reply by xzbdmw** [reply]
As I add some print, range information is not passed to preview functions, so I have to somehow write some glue code.

**Reply by ibhagwan** [reply]
You’re correct, for what you want to achieve you’ll have to save metadata somewhere, it’s not overly complex but we’re still dealing with an external terminal process (fzf) which has awareness about the selected lines only.

**Comment by ibhagwan** [comment]
Builtin preview entry point:
https://github.com/ibhagwan/fzf-lua/blob/743647f639a83e41e283d2d7daa03a85e1fbf951/lua/fzf-lua/previewer/builtin.lua#L288-L294

**Reply by xzbdmw** [reply]
Succeed! final question, how to trigger preview function if I don't want to show preview?

**Reply by ibhagwan** [reply]
Take a look at how the live color scheme preview works with `colorschemes`, the fzf previewer is set to `nohidden:0%` (otherwise the callback isn’t triggered), and the preview action is set to:
https://github.com/ibhagwan/fzf-lua/blob/743647f639a83e41e283d2d7daa03a85e1fbf951/lua/fzf-lua/providers/colorschemes.lua#L50-L66
